lib/sipPeer.js

```diff
@@ -78,5 +78,41 @@ SipPeer.prototype.moveTns = function(numbers, callback){
   this.client.makeRequest("post", url, {sipPeerTelephoneNumbers: {fullNumber: numbers}}, callback);
 };
 
+SipPeer.prototype.listApplications = function(callback) {
+  var url = this.client.concatAccountPath(SITE_PATH + "/" + this.siteId + "/" + SIP_PEER_PATH + "/" + this.id + "/" + "products" + "/" + "messaging" + "/" + "applicationSettings");
+  this.client.makeRequest("get", url, function (err, results) {
+    if (err) {
+      callback(error);
+    }
+    else {
+      var apps = results.applicationSettings;
+      if (!Array.isArray(apps.httpMessagingV2AppId) && apps.httpMessagingV2AppId) {
+        apps.httpMessagingV2AppId = [apps.httpMessagingV2AppId]
+      }
+      callback(null, apps)
+    }
+  })
+}
+
+SipPeer.prototype.editApplications = function(appData, callback) {
+  var url = this.client.concatAccountPath(SITE_PATH + "/" + this.siteId + "/" + SIP_PEER_PATH + "/" + this.id + "/" + "products" + "/" + "messaging" + "/" + "applicationSettings");
+  const data = {applicationsSettings: appData}
+  this.client.makeRequest("put", url, data, function (err, results) {
+    if (err) {callback(error);}
+    else {
+      var apps = results.applicationSettings;
+      if (!Array.isArray(apps.httpMessagingV2AppId)) {
+        apps.httpMessagingV2AppId = [apps.httpMessagingV2AppId]
+      }
+      callback(null, apps)
+    }
+  });
+}
+
+SipPeer.prototype.removeApplications = function(callback) {
+  var url = this.client.concatAccountPath(SITE_PATH + "/" + this.siteId + "/" + SIP_PEER_PATH + "/" + this.id + "/" + "products" + "/" + "messaging" + "/" + "applicationSettings");
+  const data = {applicationsSettings: 'REMOVE'}
+  this.client.makeRequest("put", url, data, callback);
+}
```

Comment:
Not a review comment, but looking at this makes me realize this functionality on this endpoint doesn't make sense. Is this removing all applications? There's no identifier to remove a specific application... It's just weird to me 🤷

Reply:
```
There is no way to assign multiple messaging applications to one sippeer so what you are removing is currently assigned application. You can find it by executing GET  to that same url
```

Apparently this is why

Reply:
Oh. I actually didn't know that. I've been treating it as an array because I thought there were multiple. 
I'll change it.

Reply:
Yeah I don't think this endpoint was well documented in our docs... I probably did the same thing on the php and ruby sdks

Reply:
I've actually gone ahead and made all three functions singular, since that's how it works (I didn't know that either and I doubt customers would).